### ISSUE TITLE
[IOS]: The plugin doesn't start to record a location when the device is moving and the app is terminated

### ISSUE THREAD

**Issue by dmitrycoder** [transistorsoft/react-native-background-geolocation#1432]
* Plugin version: 4.4.4
* Platform: iOS
* OS version: 15.1.1
* Device manufacturer / model: iPhone 12
* React Native version (`react-native -v`): 0.66.1
* Plugin config
```
    const config = {
      locationAuthorizationRequest: 'Always',
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      distanceFilter: 10,
      stopTimeout: 4,
      showsBackgroundLocationIndicator: false,
      stationaryRadius: 200,
      preventSuspend: false,
      heartbeatInterval: 120,
      enableHeadless: true,
      foregroundService: true,
      stopOnTerminate: false,
      startOnBoot: true,
      debug: false,
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      maxDaysToPersist: 2,
      url: `${BASE_URL}${API.ROUTE}`,
      headers: {
        Authorization: `Bearer ${accessToken}`,
      },
      autoSync: true,
    };
    
    const state = await BackgroundGeolocation.getState();

    if (!state.enabled) {
      BackgroundGeolocation.setConfig(config, (state) => {
        if (!state.enabled) {
          BackgroundGeolocation.start();
        }
      });
    } else {
      BackgroundGeolocation.ready(config, (state) => {
        if (!state.enabled) {
          BackgroundGeolocation.start();
        }
      });
    }
```

## Expected Behavior
When the device is moving the plugin re-awake and starting record a location.

## Actual Behavior
The plugin doesn't re-awake and not starting record a location when the app is terminated and the device is moving more 200 meters and and if the app is not being used for a while (more then 1 hour about).

## Steps to Reproduce
1. Open the app
2. Terminate the app
3. In one hour or more start moving

## Context
In the ios emulator the app works perfectly, both when it is in the foreground and in the background.
On a real physical device the plugin works perfect only when the app is in the foreground or if I terminated the app and start moving at once.
If preventSuspend set true the plugin works exactly in the same way as false.
If the debug param is true location tracking will work correctly all the time, the plugin re-awake and starting record locations even I haven't opened the app 2 and more days.

## Debug logs
<details><summary>Logs</summary>
``` 
2022-01-09 16:34:56.631647+0300 MyApp[20034:9826387] ℹ️-[TSDBLogger db_delete] maxAge: 259200
2022-01-09 16:34:56.632139+0300 MyApp[20034:9826395] ℹ️-[TSLocationManager init] 

╔═════════════════════════════════════════════
║ TSLocationManager (build 382)
╠══════════════════════════════════════════════

{
    activityRecognitionInterval = 10000;
    activityType = 1;
    authorization =     {
    };
    autoSync = 1;
    autoSyncThreshold = 0;
    batchSync = 0;
    debug = 0;
    desiredAccuracy = "-1";
    desiredOdometerAccuracy = 100;
    didDeviceReboot = 0;
    didLaunchInBackground = 0;
    didRequestUpgradeLocationAuthorization = 1;
    disableAutoSyncOnCellular = 0;
    disableElasticity = 0;
    disableLocationAuthorizationAlert = 0;
    disableMotionActivityUpdates = 0;
    disableStopDetection = 0;
    distanceFilter = 100;
    elasticityMultiplier = 1;
    enableTimestampMeta = 0;
    enabled = 1;
    extras =     {
    };
    geofenceInitialTriggerEntry = 1;
    geofenceProximityRadius = 2000;
    geofenceTemplate = "";
    headers =     {
        Authorization = "Bearer token";
        "Content-Type" = "application/json";
    };
    heartbeatInterval = 800;
    httpRootProperty = location;
    httpTimeout = 60000;
    iOSHasWarnedLocationServicesOff = 0;
    isFirstBoot = 0;
    isMoving = 0;
    lastLocationAuthorizationStatus = 3;
    locationAuthorizationAlert =     {
        cancelButton = Cancel;
        instructions = "To use background location, you must enable '{locationAuthorizationRequest}' in the Location Services settings";
        settingsButton = Settings;
        titleWhenNotEnabled = "Background location is not enabled";
        titleWhenOff = "Location services are off";
    };
    locationAuthorizationRequest = Always;
    locationTemplate = "";
    locationTimeout = 60;
    locationsOrderDirection = ASC;
    logLevel = 5;
    logMaxDays = 3;
    maxBatchSize = "-1";
    maxDaysToPersist = 2;
    maxRecordsToPersist = "-1";
    method = POST;
    minimumActivityRecognitionConfidence = 70;
    odometer = 0;
    params =     {
    };
    pausesLocationUpdatesAutomatically = 1;
    persistMode = 2;
    preventSuspend = 0;
    schedule =     (
    );
    schedulerEnabled = 0;
    showsBackgroundLocationIndicator = 0;
    startOnBoot = 1;
    stationaryRadius = 200;
    stopAfterElapsedMinutes = "-1";
    stopDetectionDelay = 0;
    stopOnStationary = 0;
    stopOnTerminate = 0;
    stopTimeout = 1;
    trackingMode = 1;
    url = "https://api/v1/track";
    useSignificantChangesOnly = 0;
}

2022-01-09 16:34:56.636232+0300 MyApp[20034:9826395] ℹ️-[GeofenceDAO init] CREATE TABLE IF NOT EXISTS geofences (id INTEGER PRIMARY KEY AUTOINCREMENT, identifier TEXT NOT NULL UNIQUE, latitude DOUBLE NOT NULL, sin_latitude DOUBLE NOT NULL, cos_latitude DOUBLE NOT NULL, longitude DOUBLE NOT NULL, sin_longitude DOUBLE NOT NULL, cos_longitude DOUBLE NOT NULL, radius DOUBLE NOT NULL, notifyOnEntry BOOLEAN NOT NULL DEFAULT 0, notifyOnExit BOOLEAN NOT NULL DEFAULT 0, notifyOnDwell BOOLEAN NOT NULL DEFAULT 0, loiteringDelay DOUBLE NOT NULL DEFAULT 0, extras TEXT)

2022-01-09 16:34:56.636320+0300 MyApp[20034:9826395] ℹ️-[GeofenceDAO init] CREATE index IF NOT EXISTS identifier ON geofences (identifier);CREATE index IF NOT EXISTS latitude ON geofences (latitude);CREATE index IF NOT EXISTS longitude ON geofences (longitude);CREATE index IF NOT EXISTS sin_latitude ON geofences (sin_latitude);CREATE index IF NOT EXISTS cos_latitude ON geofences (cos_latitude);CREATE index IF NOT EXISTS sin_longitude ON geofences (sin_longitude);CREATE index IF NOT EXISTS cos_longitude ON geofences (cos_longitude);

2022-01-09 16:34:57.005821+0300 MyApp[20034:9826366] 🔵-[TSLocationManager locationManager:didChangeAuthorizationStatus:] status 3

2022-01-09 16:34:57.008750+0300 MyApp[20034:9826365] 🔵-[LocationManager locationManager:didChangeAuthorizationStatus:] 3

2022-01-09 16:34:57.009944+0300 MyApp[20034:9826366] 🔵-[LocationManager locationManager:didChangeAuthorizationStatus:] 3

2022-01-09 16:34:57.010519+0300 MyApp[20034:9826378] 🔵-[BackgroundTaskManager locationManager:didChangeAuthorizationStatus:] 3

2022-01-09 16:34:57.010892+0300 MyApp[20034:9826378] 🔵-[LocationManager locationManager:didChangeAuthorizationStatus:] 3

2022-01-09 16:34:57.024790+0300 MyApp[20034:9826397] ℹ️+[LocationAuthorization run:onCancel:] status: 3

2022-01-09 16:35:03.273130+0300 MyApp[20034:9826094] [TSBackgroundFetch registerForTaskWithIdentifier: com.transistorsoft.app

2022-01-09 16:35:04.095232+0300 MyApp[20034:9826350] [TSBackgroundFetch scheduleProcessingTask] <TSBGTask identifier=com.transistorsoft.app, delay=900, periodic=1 requiresNetworkConnectivity=0 requiresExternalPower=0, enabled=1>

2022-01-09 16:35:04.115943+0300 MyApp[20034:9826094] [TSBackgroundFetch scheduleBGAppRefresh] com.transistorsoft.fetch

2022-01-09 16:35:04.143738+0300 MyApp[20034:9826391] ℹ️-[TSLocationManager log:message:] [RNBackgroundGeolocation startObserving]

2022-01-09 16:35:04.153171+0300 MyApp[20034:9826397] ℹ️-[TSLocationManager on:success:failure:] location
2022-01-09 16:35:04.153350+0300 MyApp[20034:9826378] ℹ️-[TSLocationManager on:success:failure:] motionchange
2022-01-09 16:35:04.153499+0300 MyApp[20034:9826388] ℹ️-[TSLocationManager on:success:failure:] activitychange
2022-01-09 16:35:04.153676+0300 MyApp[20034:9826350] ℹ️-[TSLocationManager on:success:failure:] heartbeat
2022-01-09 16:35:04.153902+0300 MyApp[20034:9826397] ℹ️-[TSGeofenceManager onGeofence:]
2022-01-09 16:35:04.154023+0300 MyApp[20034:9826388] ℹ️-[TSGeofenceManager onGeofencesChange:]
2022-01-09 16:35:04.154366+0300 MyApp[20034:9826350] ℹ️-[TSLocationManager on:success:failure:] http
2022-01-09 16:35:04.154446+0300 MyApp[20034:9826350] ℹ️-[TSLocationManager on:success:failure:] providerchange
2022-01-09 16:35:04.154726+0300 MyApp[20034:9826388] ℹ️-[TSLocationManager on:success:failure:] schedule
2022-01-09 16:35:04.155045+0300 MyApp[20034:9826350] ℹ️-[TSLocationManager on:success:failure:] powersavechange
2022-01-09 16:35:04.155367+0300 MyApp[20034:9826350] ℹ️-[TSHttpService onConnectivityChange:]
2022-01-09 16:35:04.155619+0300 MyApp[20034:9826388] ℹ️-[TSLocationManager on:success:failure:] enabledchange
2022-01-09 16:35:04.155682+0300 MyApp[20034:9826388] ℹ️-[TSHttpService onAuthorization:]
2022-01-09 16:35:04.156145+0300 MyApp[20034:9826350] 🔵-[TSLocationManager getCurrentPosition:]
2022-01-09 16:35:04.157032+0300 MyApp[20034:9826397] 🎾-[LocationManager startUpdatingLocation] ON
2022-01-09 16:35:04.164263+0300 MyApp[20034:9826397] 
📍<+50.91216,+-92.0053409> +/- 14.28m (speed -1.00 mps / course -1.00) @ 9.01.22, 16:34:51 Moscow Standard Time
2022-01-09 16:35:04.164291+0300 MyApp[20034:9826350] 🔴-[LocationManager stopUpdatingLocation] OFF
2022-01-09 16:35:04.164351+0300 MyApp[20034:9826397] 

╔═══════════════════════════════════════════════════════════
║ -[LocationManager locationManager:didUpdateLocations:] Sample 1 of 1
╚═══════════════════════════════════════════════════════════

2022-01-09 16:35:04.164525+0300 MyApp[20034:9826395] 🔵-[TSLocationManager calculateMedianLocationAccuracy:] Median location accuracy: 14.3
2022-01-09 16:35:04.169904+0300 MyApp[20034:9826397] ℹ️-[TSConfig persist]
2022-01-09 16:35:04.171045+0300 MyApp[20034:9826395] ℹ️-[TSConfig persist]
2022-01-09 16:35:04.171860+0300 MyApp[20034:9826350] 🔵-[TSLocationManager ready]
2022-01-09 16:35:04.172313+0300 MyApp[20034:9826395] ℹ️-[TSLocationManager doStart:] trackingMode: 1
2022-01-09 16:35:04.172872+0300 MyApp[20034:9826369] [javascript] '[watchPositionSuccess] -', { coords: 
   { altitude: 236.2,
     latitude: 50.9121613114,
     heading: -1,
     altitude_accuracy: 6.1,
     heading_accuracy: -1,
     accuracy: 14.3,
     speed_accuracy: -1,
     longitude: -92.0053408749141,
     speed: -1,
     floor: -1 },
  timestamp: '2022-01-09T13:34:51.440Z',
  extras: { getCurrentPosition: true },
  odometer: 0,
  uuid: '2C3EC62B-41B3-42A5-81DA-EA9016F4153A',
  is_moving: false,
  activity: { confidence: 100, type: 'unknown' },
  battery: { is_charging: true, level: 0.6399999856948853 } }
2022-01-09 16:35:04.173851+0300 MyApp[20034:9826395] 🎾-[TSGeofenceManager start]
2022-01-09 16:35:04.173875+0300 MyApp[20034:9826378] 🎾-[SOMotionDetector startDetection]
2022-01-09 16:35:04.173998+0300 MyApp[20034:9826395] 🔵-[TSLocationManager setPace:] 0
2022-01-09 16:35:04.178507+0300 MyApp[20034:9826350] ✅-[SOMotionDetector startDetection]_block_invoke Enabled M7 MotionActivity updates
2022-01-09 16:35:04.226165+0300 MyApp[20034:9826494] 🎾-[TSLocationManager startUpdatingLocation] Location-services: ON
2022-01-09 16:35:04.229125+0300 MyApp[20034:9826494] ℹ️+[LocationAuthorization run:onCancel:] status: 3
2022-01-09 16:35:04.234625+0300 MyApp[20034:9826397] 

╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:] 
╚═══════════════════════════════════════════════════════════

2022-01-09 16:35:04.234833+0300 MyApp[20034:9826378] ✅-[LocationDAO unlock]_block_invoke UNLOCKED ALL RECORDS
2022-01-09 16:35:04.235042+0300 MyApp[20034:9826494] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 1
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:04.235586+0300 MyApp[20034:9826494] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager createMotionTypeChangedHandler]_block_invoke | still/100 | isMoving: 0
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:04.236992+0300 MyApp[20034:9826494] ℹ️+[LocationAuthorization run:onCancel:] status: 3
2022-01-09 16:35:04.237212+0300 MyApp[20034:9826494] 🎾-[TSLocationManager startMonitoringBackgroundFetch] BackgroundFetch: ON
2022-01-09 16:35:04.238183+0300 MyApp[20034:9826494] 
📍<+50.91216,+-92.0053409> +/- 14.28m (speed -1.00 mps / course -1.00) @ 9.01.22, 16:35:04 Moscow Standard Time
2022-01-09 16:35:04.238267+0300 MyApp[20034:9826378] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager locationManager:didUpdateLocations:] Enabled: 1 | isMoving: 0 | df: -1.0m | age: 0.1s
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:04.238298+0300 MyApp[20034:9826494] ✅-[TSLocationManager locationManager:didUpdateLocations:] Acquired motionchange position: <+50.91216,+-92.0053409> +/- 14.28m (speed -1.00 mps / course -1.00) @ 9.01.22, 16:35:04 Moscow Standard Time
2022-01-09 16:35:04.238387+0300 MyApp[20034:9826494] 🔵-[TSLocationManager startMonitoringStationaryRegion:radius:] Radius: 200
2022-01-09 16:35:04.239681+0300 MyApp[20034:9826350] 🔴-[TSLocationManager stopUpdatingLocation]
2022-01-09 16:35:04.240276+0300 MyApp[20034:9826378] 🔵-[TSLocationManager calculateMedianLocationAccuracy:] Median location accuracy: 14.3
2022-01-09 16:35:04.240689+0300 MyApp[20034:9826378] 🎾-[TSLocationManager startMonitoringSignificantLocationChanges]
2022-01-09 16:35:04.241249+0300 MyApp[20034:9826350] ✅-[TSLocationManager persistLocation:]_block_invoke INSERT: 23454342-005B-42CC-8C82-AFAB271E93A6
2022-01-09 16:35:04.241272+0300 MyApp[20034:9826388] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:] 
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:04.274579+0300 MyApp[20034:9826494] ✅-[BackgroundTaskManager createBackgroundTask] 10
2022-01-09 16:35:04.275565+0300 MyApp[20034:9826395] ✅-[TSHttpService schedulePost] LOCKED: 23454342-005B-42CC-8C82-AFAB271E93A6
2022-01-09 16:35:04.560607+0300 MyApp[20034:9826391] 🔵-[HttpResponse handleResponse] Response: 200
2022-01-09 16:35:04.561581+0300 MyApp[20034:9826395] ✅-[TSHttpService post:]_block_invoke DESTROY: 23454342-005B-42CC-8C82-AFAB271E93A6
2022-01-09 16:35:04.561748+0300 MyApp[20034:9826395] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 1
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:04.561924+0300 MyApp[20034:9826494] ✅-[BackgroundTaskManager stopBackgroundTask:]_block_invoke 10 OF (
    10
)
2022-01-09 16:35:05.300338+0300 MyApp[20034:9826388] [connection] nw_resolver_start_query_timer_block_invoke [C19] Query fired: did not receive all answers in time for graph.facebook.com:443
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
2022-01-09 16:35:07.209753+0300 MyApp[20034:9826507] 🎾-[LocationManager watchPosition:] START
2022-01-09 16:35:07.212754+0300 MyApp[20034:9826507] ℹ️+[LocationAuthorization run:onCancel:] status: 3
2022-01-09 16:35:07.215340+0300 MyApp[20034:9826507] ✅-[BackgroundTaskManager createBackgroundTask] 11
2022-01-09 16:35:07.215475+0300 MyApp[20034:9826494] 🎾-[LocationManager onWatchPositionTimer:] ONESHOT
2022-01-09 16:35:08.704688+0300 MyApp[20034:9826499] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager createMotionTypeChangedHandler]_block_invoke | still/100 | isMoving: 0
╚═══════════════════════════════════════════════════════════
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
2022-01-09 16:35:09.231136+0300 MyApp[20034:9826499] ⚠️-[LocationManager onTimeout:] Using last-known location <+50.91216,+-92.0053409> +/- 14.28m (speed -1.00 mps / course -1.00) @ 9.01.22, 16:34:51 Moscow Standard Time
2022-01-09 16:35:09.231419+0300 MyApp[20034:9826499] ✅-[LocationManager onTimeout:] Returning best available location sample
2022-01-09 16:35:09.341573+0300 MyApp[20034:9826499] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager createMotionTypeChangedHandler]_block_invoke | still/100 | isMoving: 0
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:17.222552+0300 MyApp[20034:9826397] 
📍<+50.91216,+-92.0053409> +/- 14.28m (speed -1.00 mps / course -1.00) @ 9.01.22, 16:35:13 Moscow Standard Time
2022-01-09 16:35:17.224240+0300 MyApp[20034:9826367] 🔴-[LocationManager stopUpdatingLocation] OFF
2022-01-09 16:35:17.225128+0300 MyApp[20034:9826397] 
╔═══════════════════════════════════════════════════════════
║ -[LocationManager locationManager:didUpdateLocations:] Sample 1 of 1
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:17.225281+0300 MyApp[20034:9826391] 🔵-[TSLocationManager calculateMedianLocationAccuracy:] Median location accuracy: 14.3
2022-01-09 16:35:17.227151+0300 MyApp[20034:9826369] [javascript] '[watchPositionSuccess] -', { coords: 
   { altitude: 236.2,
     latitude: 50.9121613114,
     heading: -1,
     altitude_accuracy: 5.7,
     heading_accuracy: -1,
     accuracy: 14.3,
     speed_accuracy: -1,
     longitude: -92.0053408749141,
     speed: -1,
     floor: -1 },
  timestamp: '2022-01-09T13:35:13.295Z',
  extras: { watch: true },
  odometer: 0,
  uuid: '10B7D6D5-DEFA-4F96-8F85-5EAB6B2D0BCB',
  is_moving: false,
  activity: { confidence: 100, type: 'still' },
  battery: { is_charging: true, level: 0.6399999856948853 } }
2022-01-09 16:35:17.231445+0300 MyApp[20034:9826367] ✅-[TSLocationManager persistLocation:]_block_invoke INSERT: 10B7D6D5-DEFA-4F96-8F85-5EAB6B2D0BCB
2022-01-09 16:35:17.231564+0300 MyApp[20034:9826397] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:] 
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:17.234905+0300 MyApp[20034:9826391] ✅-[BackgroundTaskManager createBackgroundTask] 12
2022-01-09 16:35:17.237271+0300 MyApp[20034:9826391] ✅-[TSHttpService schedulePost] LOCKED: 10B7D6D5-DEFA-4F96-8F85-5EAB6B2D0BCB
2022-01-09 16:35:17.492772+0300 MyApp[20034:9826391] 🔵-[HttpResponse handleResponse] Response: 200
2022-01-09 16:35:17.499184+0300 MyApp[20034:9826570] ✅-[TSHttpService post:]_block_invoke DESTROY: 10B7D6D5-DEFA-4F96-8F85-5EAB6B2D0BCB
2022-01-09 16:35:17.499343+0300 MyApp[20034:9826570] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 1
╚═══════════════════════════════════════════════════════════
2022-01-09 16:35:17.499434+0300 MyApp[20034:9826570] ✅-[BackgroundTaskManager stopBackgroundTask:]_block_invoke 12 OF (
    11,
    12
)
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
2022-01-09 16:35:19.537521+0300 MyApp[20034:9826367] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager createMotionTypeChangedHandler]_block_invoke | still/100 | isMoving: 0
╚═══════════════════════════════════════════════════════════
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
2022-01-09 16:35:27.504030+0300 MyApp[20034:9826350] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager createMotionTypeChangedHandler]_block_invoke | still/100 | isMoving: 0
╚═══════════════════════════════════════════════════════════
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
2022-01-09 16:35:32.478877+0300 MyApp[20034:9826697] 🔴-[LocationManager stopWatchPosition] STOP
2022-01-09 16:35:32.480247+0300 MyApp[20034:9826698] ✅-[BackgroundTaskManager stopBackgroundTask:]_block_invoke 11 OF (
    11
)
2022-01-09 16:35:33.379163+0300 MyApp[20034:9826567] 🔵-[TSLocationManager onSuspend:] enabled? 1)
2022-01-09 16:35:33.388324+0300 MyApp[20034:9826567] ℹ️-[TSDBLogger db_save] Log committed
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
Message from debugger: Terminated due to signal 9
```
</details>

**Comment by christocracy** [comment]
I test the *SampleApp* for *Cordova*, *Capacitor*, *React Native* and *Flutter* on my device, all configured with `debug: false`, running untouched in the background for months at a time.  They never fail to track me.

`Config.debug` has no effect on performance.

**Comment by dmitrycoder** [comment]
I believe you of course) I just try to understand my case. Why do if I start the app and then kill(terminated) it and after that I go for a walk at once and when I'm moving more then 200 meters approximately the plugin starts to record a location. Otherwise If I will do the same but go for a walk in 30 minutes or more the plugin doesn't re-awake.

I try fix it 2 months. It seems I already reread the documentation a lot of times and checked all cases but I see the same problem all the time. So far `debug: true` only helped me. Maybe it because push notifications re-awake the app or the plugin maybe something else I don't know.

The plugin on android works very nice. Unfortunately my application is cross platforms. There is no sense if that will be working on android only.

**Comment by christocracy** [comment]
> Maybe it because push notifications re-awake the app

They are not “push” notifications, they are “local” notifications.  These debug notifications are emitted only when the app is already alive.  

I suggest you install the SampleApp and try it.  The *Advanced App* has a settings screen where you can turn off *Debug* mode with a switch.

**Comment by dmitrycoder** [comment]
> Your code looks messed up. You **must** call `.ready()` every single time your app launches, no matter what. Not just _sometimes_.
> 
> ```js
> if (!state.enabled) {
>       BackgroundGeolocation.setConfig(config, (state) => {
>         if (!state.enabled) {
>           BackgroundGeolocation.start();
>         }
>       });
>     } else {
>       BackgroundGeolocation.ready(config, (state) => {
>         if (!state.enabled) {
>           BackgroundGeolocation.start();
>         }
>       });
>     }
> ```

That's because I call BackgroundGeolocation.ready on location onboarding screen and then after auth I set BackgroundGeolocation.setConfig. But if user already signed in and start app I call BackgroundGeolocation.ready every single time the app launches.

Now it looks like this

`
if (isAfterAuth) {
      BackgroundGeolocation.setConfig(config, (state) => {
        if (!state.enabled) {
          BackgroundGeolocation.start();
        }
      });
    } else {
      BackgroundGeolocation.ready(config, (state) => {
        if (!state.enabled) {
          BackgroundGeolocation.start();
        }
      });
    }
 `

**Comment by dmitrycoder** [comment]
> I suggest you install the SampleApp and try it. The _Advanced App_ has a settings screen where you can turn off _Debug_ mode with a switch.

I installed _Advanced App_ and turn off _Debug_ mode and terminated the app. On the next day I go to my business and there are no tracks in http://tracker.transistorsoft.com/mycompany only couple of them when I opened the app they send to your server.
But my application tracked all my locations the same moment.
Then I removed the _Advanced App_ and start moving. After that my application nothing tracked.
How is it possible?
If two and more applications with the plugin installed on iphone will work only one?
Otherwise how could the _Advanced App_ have influenced on my app?

**Comment by christocracy** [comment]
> If two and more applications with the plugin installed on iphone will work only one?

Apps do not affect each other.  I test 4 apps simultaneously, for React Native, Fluttter, Cordova and Capacitor.  All with debug: false.  None fail to track me (for years).

**Comment by robertoaltieri** [comment]
We are seeing the same issue with an app that works well on previous versions of the OS. With iOS 15 the recording works intermittently.

**Comment by dmitrycoder** [comment]
> We are seeing the same issue with an app that works well on previous versions of the OS. With iOS 15 the recording works intermittently.

I agree with you. I checked it and I see on iPhones with iOS < 15 the plugin works fine.

**Comment by christocracy** [comment]
Ensure you are in latest release of iOS.  There are many reports from pure iOS developers of a bug introduces into iOS Signifidant Location changes and geofencing api.

I experience no issues here with the SampleApp which is field-tested daily.

**Comment by dmitrycoder** [comment]
I've tested the SampleApp(Advanced) with default configuration (only `debug: false`) on my iPhone 12 with latest release of iOS 15.2.1. 
How you can see on the screenshot.
Yesterday right after installed the app I switched on the play button(changePace: true). 
Today I gone to work. When I came to work I open the app and switch off the play button(changePace: false). 
After working day I go home and when I came home I open the app and see nothing only one track because I opened the app. I see the same behaviour on my application all the time => _The plugin doesn’t work on iOS >= 15 when the application is terminated._

![https:tracker transistorsoft com png 2022-01-24 21-28-21](https://user-images.githubusercontent.com/24842537/150842933-b881356a-d2df-48e1-b0c8-893a5345e193.png)

**Comment by murtazakaz** [comment]
I read all the above messages and mine is quite similar.

This same case is happening with me in ios > OS 15, tracking not working smoothly like it was before. 
It stopped or terminated in the background resulting in missing GPS and no tracking.
Sometimes when I'm moving, the plugin did not start or send any GPS events. I logged this in file and didn't see any event from your plugin.

Is there any way to log your plugin in some file so I can see what's happening in the background release mode and why the app is being terminated or stop sending GPS data in the background?

Please see both log files that I created for this purpose:

First log attachment, when app works fine in the background:
[allegory_log_file_2022-01-21T17:57:19.013Z 2.txt](https://github.com/transistorsoft/react-native-background-geolocation/files/7933096/allegory_log_file_2022-01-21T17.57.19.013Z.2.txt)

Second log attachment, I was driving but tracking did not start and no event logged there:
[allegory_log_file_2022-01-23T14_37_23.208Z.txt](https://github.com/transistorsoft/react-native-background-geolocation/files/7933097/allegory_log_file_2022-01-23T14_37_23.208Z.txt)

**Comment by christocracy** [comment]
There are many reports from all over the development world of [iOS Geofencing Not Working](https://www.google.com/search?q=ios+15+geofencing+not+working&sxsrf=AOaemvKxLmrz8uDpyqgH4QYa_FNBkCqGhA%3A1643116919607&ei=d_nvYYe4JIqT0PEPiPSRKA&ved=0ahUKEwjHz8TC_8z1AhWKCTQIHQh6BAUQ4dUDCA4&uact=5&oq=ios+15+geofencing+not+working&gs_lcp=Cgdnd3Mtd2l6EAMyBAgjECc6BwgjELADECc6BwgAEEcQsAM6BwgjELACECc6BggAEAcQHkoECEEYAEoECEYYAFDqA1jeCGC2CmgBcAJ4AIAB3AGIAYkGkgEFMC4zLjGYAQCgAQHIAQTAAQE&sclient=gws-wiz).  The plugin relies upon a geofence exit event to detect when a device is moving (and launch the app in the background if the app is terminated).

Some sources say the bug is fixed in [iOS 15.3 beta](https://blog.psngr.co/ios-15-geofencing-problem/)

**Comment by murtazakaz** [comment]
@christocracy thanks for the reply, I just want to know about one thing,

Is there any way to write all logs from your plugin in some file (in the same way as in Xcode) so I can see what's happening in the background release mode and when the app is being terminated or stop sending GPS data in the background?

**Comment by christocracy** [comment]
> Is there any way to write all logs from your plugin 

The plugin is already doing that.  All logs are saved to its own SQLite db.  See api docs “Logger”.

**Comment by murtazakaz** [comment]
> > Is there any way to write all logs from your plugin
> 
> The plugin is already doing that. All logs are saved to its own SQLite db. See api docs “Logger”.

thanks

**Comment by murtazakaz** [comment]
@christocracy Just want to share the logs from last night's trip, was it the same geofencing issue or something else?
[
![gpsTrackingMissingTime_1](https://user-images.githubusercontent.com/18357274/151156034-dbfe64a0-9f86-4006-b05b-ef1c9879aefa.png)
](url)

**Comment by christocracy** [comment]
I have no idea what you’re trying to show me with those logs and I have no idea what happened “last night”.

**Comment by murtazakaz** [comment]
> I have no idea what you’re trying to show me with those logs and I have no idea what happened “last night”.

Our app works in the background and wakes up whenever a significant location change occurs with a driving event. With that, we start collecting GPS and sensor data per second. In the screenshot shared with you: a driving event started at 2022-01-26 00:20:10.027 and ended around 2022-01-26 00:30:00.000. However, the app/plugin stopped receiving any location change at 2022-01-26 00:31:49.079, although the user was in a car and driving at least 20 km/h. After the driving event ended and the user was in still/not moving mode, the plugin woke up at 2022-01-26 00:31:49.079 and worked until 2022-01-26 00:31:57.033. After this moment, until the app was reopened again at 2022-01-26 07:51:33.503, the plugin didn’t end the trip event.

**Comment by christocracy** [comment]
@murtazakaz you've hijacked someone else's issue and not provided any background information.  I suggest you post a new issue.

<img width="863" alt="Screen Shot 2022-01-26 at 9 06 29 AM" src="https://user-images.githubusercontent.com/14999/151177804-3bc814a2-4b2b-472d-9053-256ecdf3b548.png">

**Comment by dmitrycoder** [comment]
> There are many reports from all over the development world of [iOS Geofencing Not Working](https://www.google.com/search?q=ios+15+geofencing+not+working&sxsrf=AOaemvKxLmrz8uDpyqgH4QYa_FNBkCqGhA%3A1643116919607&ei=d_nvYYe4JIqT0PEPiPSRKA&ved=0ahUKEwjHz8TC_8z1AhWKCTQIHQh6BAUQ4dUDCA4&uact=5&oq=ios+15+geofencing+not+working&gs_lcp=Cgdnd3Mtd2l6EAMyBAgjECc6BwgjELADECc6BwgAEEcQsAM6BwgjELACECc6BggAEAcQHkoECEEYAEoECEYYAFDqA1jeCGC2CmgBcAJ4AIAB3AGIAYkGkgEFMC4zLjGYAQCgAQHIAQTAAQE&sclient=gws-wiz). The plugin relies upon a geofence exit event to detect when a device is moving (and launch the app in the background if the app is terminated).
> 
> Some sources say the bug is fixed in [iOS 15.3 beta](https://blog.psngr.co/ios-15-geofencing-problem/)

@christocracy hello. You were right! The bug was fixed on iOS 15.3. Now the plugin works perfectly! Thank you so much. You can close this issue 🙂
